common/logisticspipes/logistics/LogisticsFluidManager.java

```diff
@@ -28,33 +32,41 @@
 public class LogisticsFluidManager implements ILogisticsFluidManager {
 
 	@Override
-	public Pair<Integer, Integer> getBestReply(FluidIdentifierStack stack, IRouter sourceRouter, List<Integer> jamList) {
-		for (ExitRoute candidateRouter : sourceRouter.getIRoutersByCost()) {
-			if (!candidateRouter.containsFlag(PipeRoutingConnectionType.canRouteTo)) {
-				continue;
-			}
-			if (candidateRouter.destination.getSimpleID() == sourceRouter.getSimpleID()) {
-				continue;
+	public Pair<Integer, FluidSinkReply> getBestReply(FluidIdentifierStack stack, IRouter sourceRouter, List<Integer> jamList) {
+		final Integer[] resultRouterId = { null };
+		final FluidSinkReply[] result = { null };
```

Comment:
reduce the stream instead of using side-effects

Reply:
I need help with this because I have absolutely no idea how to use reduce() in this case. Ive doctored around with it quite a but but to no avail.

Reply:
I changed this by mapping the qualified possible results into the pairs I need anyways and used reduce on those. Gotta say the result is a lot prettier!